Calculator.rb

```diff
@@ -0,0 +1,97 @@
+# This is for the Calculator exercise, the first exercise we turn in!
+# Specifics: add, subtract, multiply and divide using two numbers
+# Silver: Print out the formula in addition to the result, i.e. 2 + 2 = 4
+  # Add support for computing exponents (2^4 = 2 * 2 * 2 * 2 = 16).
+  # Add support for the modulo operator (10 % 3 = 1).
+# Gold Gracefully handle unexpected user input:
+
+puts "Hello and welcome to Calculator."
+puts "Please specify what type of math you'd like to do:"
+
+# This is what type of calculation will be performed
+math_operation = gets.chomp.downcase
+
+# if the variable is empty, run this until it contains something
+until math_operation != ""
+  puts "You have typed in nothing. Please select from +, -, /, *, %"
+  math_operation = gets.chomp.downcase
+end
+
+# Checking to see if what user typed is included in library of valid math operations
+maths = ["+", "add", "addition","sub", "subtract", "subtraction", "-",
+   "/", "div", "divide", "division", "*", "mul", "multiply",
+   "^", "exp", "exponent", "%", "mod", "modulo", "modulus"]
+
+if maths.include?(math_operation)
+else
+  puts "Please type in a operation: "
+  math_operation = gets.chomp
+end
+
```

Comment:
What happens when you type in an operation that is invalid after line 27? 

I think that since this is not included within a while or until block it will store that value. If you make sure that it is in the maths array in a loop you'll be able to avoid the ending else.